### PR TITLE
change anonymous applicant to guest

### LINKED ISSUE #1918: In applicant experience, change language from "Anonymous Applicant" to "Guest"
Feedback that "Anonymous Applicants" is a strange term and "Guest" would be preferred.

### LINKED ISSUE #2354: "Guest" is not localized
**Describe the bug**
For Guest applicants the word "Guest" is not localized.

**To Reproduce**
1. Login as Guest
2. Pick a non-English language
3. Upper right corner says "Guest"

**Expected behavior**
"Guest" should be translated to the language that user chose. 

**Screenshots**
Screen Shot 2022-04-26 at 11 42 46 AM

value coming from https://github.com/seattle-uat/civiform/blob/401d969fe9d096b5391ee85feca2124a87b0301b/universal-application-tool-0.0.1/app/services/applicant/ApplicantServiceImpl.java#L328

### PR DESCRIPTION
**Pull request by azizova-leyla**
### Description
Changing ApplicantName to be Optional<String> and resolving it in the views/controllers instead to allow for i18n. 

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)
Fixes #1918, Fixes #2354 
Screen Shot 2022-04-26 at 11 42 46 AM

### CODE REVIEW

**Review by clouser2 — DISMISSED** [review]

File: universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java [hunk 1]
```diff
@@ -74,7 +74,7 @@ public String getApplicantName() {
       return firstName;
     } catch (NoSuchElementException e) {
       logger.error("Application {} does not include an applicant name.");
-      return "<Anonymous Applicant>";
+      return "Guest";
```
Comment (bion) [inline universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java:77]: suggestion (blocking): please use the messages API and add "Guest" as a string in the `us-EN` locale

**Review by bion — CHANGES_REQUESTED** [review]

File: universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java (hunk 1 above)
Comment (bion) [reply universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java:77]: https://docs.civiform.us/contributor-guide/developer-guide/internationalization-i18n

**Review by clouser2 — DISMISSED** [review]

File: universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java (hunk 1 above)
Comment (clouser2) [reply universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java]: @azizova-leyla and I addressed this in the latest commits.

File: server/app/views/applicant/ApplicantProgramBlockEditView.java [hunk 2]
```diff
@@ -92,6 +94,7 @@ public Content render(Params params) {
     bundle.addFooterScripts(layout.viewUtils.makeLocalJsTag("validation"));
 
     return layout.renderWithNav(
+        // TODO(clouser): Make sure applicant name is resolved.
```
Comment (bion) [inline server/app/views/applicant/ApplicantProgramBlockEditView.java:97]: issue (blocking): please expand on this or remove

**Review by bion — APPROVED** [review]
Thanks for sticking with this and making the deeper improvement!

File: server/app/views/applicant/ApplicantProgramBlockEditView.java (hunk 2 above)
Comment (azizova-leyla) [reply server/app/views/applicant/ApplicantProgramBlockEditView.java]: oops, that's fixed. Removing.

**Review by bion — APPROVED** [review]

**Review by clouser2 — APPROVED** [review]

**Pull request merged**